app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadViewModel.kt

```diff
@@ -73,6 +73,7 @@ class ThreadViewModel @Inject constructor(
 
     val threadLive = MutableLiveData<Thread?>()
     val messagesLive = MutableLiveData<Pair<ThreadAdapterItems, MessagesWithoutHeavyData>>()
+    val batchedMessages = SingleLiveEvent<List<Any>>()
```

Comment:
Instead of having to rely in SingleLiveEvent can't we have a live data that represents the list of all loaded messages? This way we can listen in a straight forward manner to the list of messages, when the list updates, we submit the new list. The logic to delay the loading can then be handle on its own independently and the logic is not split depending on if we have more or less messages than `ThreadViewModel.SUPER_COLLAPSED_BLOCK_MINIMUM_MESSAGES_LIMIT`

Reply:
This will just complicated the code.
It means adding a new LiveData used only to `submitList()` to the Adapter, when we can directly `submitList()` now.

The logic to delay is already handled on its own independently, in `displayBatchedMessages()`.

Reply:
1) Directly submitting the list by hand in two places makes the coupling of the code higher where as introducing a LiveData as the only place that links the logic where data is emitted from to the logic where it's consumed makes the coupling the lowest possible.

2) Also, how does this new logic interact with the print fragment?

Reply:
Right now, I feel like this PR adds more coupling and coherence issues to the already existing coupling and coherence issue. We should quickly handle this first, before this PR imo because it doesn't seem that hard nor long to fix. 

I think we should fix this before it gets uncontrollable